src/rendering/RenderUtil.cc

```diff
@@ -2058,6 +2058,27 @@ void RenderUtilPrivate::LowlightNode(const rendering::NodePtr &_node)
   }
 }
 
+/////////////////////////////////////////////////
+void RenderUtil::ScaleWireBox(const Entity &_entityId,
+    const math::Vector3d &_scale)
+{
+  if (this->dataPtr->wireBoxes.find(_entityId) ==
+          this->dataPtr->wireBoxes.end())
+  {
+    return;
+  }
+
+  ignition::rendering::WireBoxPtr wireBox = this->dataPtr->wireBoxes[_entityId];
+  if (!wireBox)
+    return;
+
+  auto visParent = wireBox->Parent();
+  if (!visParent)
+    return;
+
+  visParent->SetLocalScale(_scale);
```

Comment:
I notice that the the wirebox is scaled correctly the first time but subsequent scales are not correct. It's likely that you'll need to multiply by the new scale instead of setting it directly.

Reply:
Fixed in 2410668 (by you :)